cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/nosql/NoSqlStructuredTable.java

```diff
@@ -126,6 +126,20 @@ public CloseableIterator<StructuredRow> scan(Range keyRange, int limit) throws I
                              limit);
   }
 
+  @Override
+  public CloseableIterator<StructuredRow> scan(Range keyRange, int limit, Field<?> filterIndex)
+    throws InvalidFieldException, IOException {
+    LOG.trace("Table {}: Scan range {} with limit {} and index {}", schema.getTableId(), keyRange, limit, filterIndex);
+    fieldValidator.validateField(filterIndex);
+    if (!schema.isIndexColumn(filterIndex.getName())) {
+      throw new InvalidFieldException(schema.getTableId(), filterIndex.getName(), "is not an indexed column");
+    }
+    LimitIterator rangeIterator = new LimitIterator(Collections.singleton(new ScannerIterator(getScanner(keyRange),
```

Comment:
nit: this is a bit hard to read, would be better to split it up and do the Collections.singleton(...) on a previous line